core/runtime/common/storage_code_provider.cpp

```diff
@@ -42,21 +43,15 @@ namespace kagome::runtime {
           return cached_code_;
         }
       }
-      auto batch = storage_->getEphemeralBatchAt(state);
-      // avoids failure in case when the state was pruned
-      if(not batch.has_value()) {
-        batch = storage_->getEphemeralBatch();
-      }
-      setStateCodeFromBatch(batch);
+      OUTCOME_TRY(batch, storage_->getEphemeralBatchAt(state));
+      setStateCodeFromBatch(*batch.get());
     }
     return cached_code_;
   }
 
   void StorageCodeProvider::setStateCodeFromBatch(
-      const outcome::result<std::unique_ptr<storage::trie::EphemeralTrieBatch>>
-          &batch) const {
-    BOOST_ASSERT_MSG(batch.has_value(), "Error getting a batch of the storage");
-    auto state_code_res = batch.value()->get(storage::kRuntimeCodeKey);
+      const storage::trie::EphemeralTrieBatch &batch) const {
+    auto state_code_res = batch.get(storage::kRuntimeCodeKey);
     BOOST_ASSERT_MSG(state_code_res.has_value(),
```

Comment:
Hmm... May it is better to return `outcome::result<void>` for error processing in calling function?